src/common/exception/src/exception_code.rs

```diff
@@ -192,6 +192,8 @@ build_exceptions! {
     DropDbWithDropTime(2315),
     UndropDbWithNoDropTime(2316),
     TxnRetryMaxTimes(2317),
+    CreateUnsupportedCatalog(2318),
```

Comment:
Keep the same error naming style.

I prefer to use `CataglogNotSupported`.

Also, please add a doc on what this error means and when to use this error.